File: backend/src/main/java/com/briefen/model/UserSettings.java
package com.briefen.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.Map;

/**
 * Stores user preferences. Since Briefen has no auth, a single
 * document with a well-known id ("default") is used.
 */
@Document(collection = "settings")
public class UserSettings {

    public static final String DEFAULT_ID = "default";

    @Id
    private String id = DEFAULT_ID;

    private String defaultLength = "default"; // "shorter" | "default" | "longer"
    private String model; // null means use server default from application.yml
    private boolean notificationsEnabled = false;
    private String openaiApiKey; // plaintext — acceptable for local single-user app
    private String readeckApiKey;
    private String readeckUrl;
    private Instant updatedAt;

    public UserSettings() {
        this.updatedAt = Instant.now();
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getDefaultLength() { return defaultLength; }
    public void setDefaultLength(String defaultLength) {
        this.defaultLength = defaultLength;
        this.updatedAt = Instant.now();
    }

    public String getModel() { return model; }
    public void setModel(String model) {
        this.model = model;
        this.updatedAt = Instant.now();
    }

    public String getOpenaiApiKey() { return openaiApiKey; }
    public void setOpenaiApiKey(String openaiApiKey) {
        this.openaiApiKey = openaiApiKey;
        this.updatedAt = Instant.now();
    }

    public boolean isNotificationsEnabled() { return notificationsEnabled; }
    public void setNotificationsEnabled(boolean notificationsEnabled) {
        this.notificationsEnabled = notificationsEnabled;
        this.updatedAt = Instant.now();
    }

    public String getReadeckApiKey() { return readeckApiKey; }
    public void setReadeckApiKey(String readeckApiKey) {
        this.readeckApiKey = readeckApiKey;
        this.updatedAt = Instant.now();
    }

    public String getReadeckUrl() { return readeckUrl; }
    public void setReadeckUrl(String readeckUrl) {
        this.readeckUrl = readeckUrl;
        this.updatedAt = Instant.now();
    }

    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}

File: mobile/lib/features/settings/domain/user_settings.dart
class UserSettings {
  final String? defaultLength;
  final String? model;
  final bool notificationsEnabled;
  final String? openaiApiKey;
  final String? anthropicApiKey;
  final String? readeckApiKey;
  final String? readeckUrl;
  final String? webhookUrl;
  final String? customPrompt;

  const UserSettings({
    this.defaultLength,
    this.model,
    this.notificationsEnabled = false,
    this.openaiApiKey,
    this.anthropicApiKey,
    this.readeckApiKey,
    this.readeckUrl,
    this.webhookUrl,
    this.customPrompt,
  });

  factory UserSettings.fromJson(Map<String, dynamic> json) {
    return UserSettings(
      defaultLength: json['defaultLength'] as String?,
      model: json['model'] as String?,
      notificationsEnabled: json['notificationsEnabled'] as bool? ?? false,
      openaiApiKey: json['openaiApiKey'] as String?,
      anthropicApiKey: json['anthropicApiKey'] as String?,
      readeckApiKey: json['readeckApiKey'] as String?,
      readeckUrl: json['readeckUrl'] as String?,
      webhookUrl: json['webhookUrl'] as String?,
      customPrompt: json['customPrompt'] as String?,
    );
  }

  UserSettings copyWith({
    Object? defaultLength = _sentinel,
    Object? model = _sentinel,
    bool? notificationsEnabled,
    Object? openaiApiKey = _sentinel,
    Object? anthropicApiKey = _sentinel,
    Object? readeckApiKey = _sentinel,
    Object? readeckUrl = _sentinel,
    Object? webhookUrl = _sentinel,
    Object? customPrompt = _sentinel,
  }) {
    return UserSettings(
      defaultLength: defaultLength == _sentinel
          ? this.defaultLength
          : defaultLength as String?,
      model: model == _sentinel ? this.model : model as String?,
      notificationsEnabled: notificationsEnabled ?? this.notificationsEnabled,
      openaiApiKey: openaiApiKey == _sentinel
          ? this.openaiApiKey
          : openaiApiKey as String?,
      anthropicApiKey: anthropicApiKey == _sentinel
          ? this.anthropicApiKey
          : anthropicApiKey as String?,
      readeckApiKey: readeckApiKey == _sentinel
          ? this.readeckApiKey
          : readeckApiKey as String?,
      readeckUrl: readeckUrl == _sentinel
          ? this.readeckUrl
          : readeckUrl as String?,
      webhookUrl: webhookUrl == _sentinel
          ? this.webhookUrl
          : webhookUrl as String?,
      customPrompt: customPrompt == _sentinel
          ? this.customPrompt
          : customPrompt as String?,
    );
  }
}

const _sentinel = Object();
